src/sagemaker/estimator.py

```diff
@@ -660,7 +667,52 @@ def __init__(
         self.checkpoint_local_path = checkpoint_local_path
 
         self.rules = rules
-        self.debugger_hook_config = debugger_hook_config
+
+        # First get the config value provided for debugger_hook_config
```

Comment:
if the config has a dict, and the user explicitly passed in False, I think this should skip the config dict. Dont think it would do that today though right? Maybe we could walk thru all the scenarios and make sure each has the behavior we want